File: backend/Models/Flight.cs
using backend.Types;

namespace backend.Models
{
    public class Flight
    {
        public Flight(Guid id, string username, Location departureLocation, Location destination, string departureDate, string arrivalDate, double price, Dictionary<string, int> addons, int[][] seats)
        {
            Id = id;
            Username = username;
            DepartureLocation = departureLocation;
            Destination = destination;
            DepartureDate = departureDate;
            ArrivalDate = arrivalDate;
            Price = price;
            Addons = addons;
            Seats = seats;
        }

        public Guid Id { get; set; }
        public string Username { get; set; }
        public Location DepartureLocation { get; set; }
        public Location Destination { get; set; }
        public string DepartureDate { get; set; }
        public string ArrivalDate {get; set;}
        public double Price { get; set; }
        public Dictionary<string, int> Addons {get; set;}
        public int[][] Seats { get; set; }
    }
}

File: frontend/src/services/flight/types/Flight.ts
export class Flight {
    public id: string;
    public username: string;
    public departureLocation: {country: string, city: string, airport: string};
    public destination: {country: string, city: string, airport: string};
    public departureDate: string;
    public arrivalDate: string;
    public price: number;
    public addons: Map<string, number>;
    public seats: number[][];

    public constructor(flight: Flight | null = null,
                       id: string = "",
                       username: string = "",
                       departureLocation: {country: string, city: string, airport: string} = {country: "", city: "", airport: ""}, 
                       destination: {country: string, city: string, airport: string} = {country: "", city: "", airport: ""},
                       departureDate: string = "",
                       arrivalDate: string = "",
                       price: number = 0,
                       addons: Map<string, number> = new Map(),
                       seats: number[][] = []) {
        if (!flight) {
            this.id = id;
            this.username = username;
            this.departureLocation = departureLocation;
            this.destination = destination;
            this.departureDate = departureDate;
            this.arrivalDate = arrivalDate;
            this.price = price;
            this.addons = addons;
            this.seats = seats;
        } else {
            this.id = flight.id;
            this.username = flight.username;
            this.departureLocation = flight.departureLocation;
            this.destination = flight.destination;
            this.departureDate = flight.departureDate;
            this.arrivalDate = flight.arrivalDate;
            this.price = flight.price;
            this.addons = flight.addons;
            this.seats = flight.seats;
        }
    }
}